### PR TITLE
resolver: Refactor and document the package

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Core changes:

- typed `DNS` resolver to follow same interface as `NNS`
- removed testnet/mainnet name formats since they are currently not documented (or ?)
- simplified interface of `NNS` resolver: now it accepts network address instead of neo client instance

### CODE REVIEW

**Comment by cthulhu-rider** [comment]
I propose to rename package since `resolver` is more appropriate for an entity not a package. Names that come to mind:

- `names`
- `nameservice`
- `ns` meaning name service, maybe too short
- `nns` meaning NeoFS name service, collides with Neo name service
- `neofsnet` similar to `net` since DNS functionality is placed in this standard package, but then I'd expect features like client to be presented inside this package

**Comment by alexvanin** [comment]
`ns` is good in my opinion.

**Comment by cthulhu-rider** [comment]
> in the code `ns.ResolveContainerName` may confuse you a little.

It's going to be `ns.DNS.ResolveContainerName` or `ns.NNS.ResolveContainerName` actually.

**Comment by alexvanin** [comment]
> simplified interface of NNS resolver: now it accepts network address instead of neo client instance

I think having an possibility to provide a client is still viable, but it can wait until we will go client pool in S3 or CDN apps.

However, consider passing websocket endpoint addresses with creating a websocket client, as it is better in our [tests](https://github.com/nspcc-dev/neofs-node/pull/1170#issuecomment-1090393003).

**Comment by cthulhu-rider** [comment]
@alexvanin @carpawell what's better in your opinion:
1. `ns.DNS` / `ns.NNS`
2. `ns.Domain` / `ns.Neo`

Thumbs up for 1 and heart for 2.

**Comment by alexvanin** [comment]
https://github.com/nspcc-dev/neofs-sdk-go/pull/179/commits/9531520548f167bb0de4f3682ab51dce19a1f9df

Oups, I meant it will be nice to have a parameter of creating WSClient or regular client. WS clients are only supported by neo-go, so we should have HTTP option for developers which may use regular Neo nodes.

I thought about three options:
1. parse `ws:` and `http[s]:` scheme from address string and choosing client type inside SDK,
2. pass address type in `Dial(address, typ)`,
3. have `Dial(address)` and `DialWS(address)` methods (similar to `New` and `NewWS` from neo-go).

**Comment by cthulhu-rider** [comment]
@alexvanin implemented through URL scheme parsing.

File: ns/nns.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	var multiSchemeClient interface {
+		neoClient
+		// Init turns client to "ready-to-work" state.
+		Init() error
+		// GetContractStateByID returns state of the NNS contract on 1 input.
+		GetContractStateByID(int32) (*state.Contract, error)
+	}
+
+	uri, err := url.Parse(address)
+	if err == nil && uri.Scheme == "ws" {
```
Comment (alexvanin) [inline ns/nns.go:72]: `wss` scheme is also valid for websocket client.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**